aea/test_tools/utils.py

```diff
@@ -61,7 +74,23 @@ def as_context(*contexts: Any) -> Generator[None, None, None]:
 
 def copy_class(cls: Type) -> Type:
     """Copy a class. Useful for testing class setup configurations"""
-    return type(cls.__name__, cls.__bases__, dict(cls.__dict__))
+
+    # NOTE: this does not recursively deepcopy the class
+    #       nor can it copy items that cannot be pickled,
+    #       e.g. classes containing classmethods in __dict__.
+    #       Use at your own risk.
```

Comment:
I think this tool can be quite useful, but a word of warning is in order. Ideally this would be a pure `deepcopy`, but that is not entirely trivial to implement so I decided to opt for `get to simple get to done` and leave a note for other devs.